mlflow/entities/document.py

```diff
@@ -0,0 +1,47 @@
+from copy import deepcopy
+from dataclasses import asdict, dataclass, field
+from typing import Dict, Optional
+
+
+@dataclass
+class Document:
+    """
+    An entity used in MLflow Tracing to represent retrieved documents in a RETRIEVER span.
+
+    Args:
+        page_content: The content of the document.
+        metadata: A dictionary of metadata associated with the document.
+        id: The ID of the document.
+    """
+
+    page_content: str
+    metadata: Dict[str, any] = field(default_factory=dict)
+    id: Optional[str] = None
+
+    @classmethod
+    def from_langchain_document(cls, document):
+        # older versions of langchain do not have the id attribute
+        id = document.id if hasattr(document, "id") else None
+
+        return cls(
+            page_content=document.page_content,
+            metadata=deepcopy(document.metadata),
+            id=id,
+        )
+
+    @classmethod
+    def from_llama_index_node_with_score(cls, node_with_score):
+        metadata = {}
+        metadata["score"] = node_with_score.get_score()
+        # update after setting score so that it can be
+        # overridden if the user wishes to do so
+        metadata.update(deepcopy(node_with_score.metadata))
```

Comment:
we can simplify this to:

```suggestion
        metadata = {
            "score": node_with_score.get_score(),
            **deepcopy(node_with_score.metadata),
        }
```